lib/handler/mruby.c

```diff
@@ -329,6 +312,24 @@ static h2o_mruby_shared_context_t *get_shared_context(h2o_context_t *ctx)
     return *data;
 }
 
+mrb_value prepare_fibers(h2o_mruby_context_t *ctx)
+{
+    mrb_state *mrb = ctx->shared->mrb;
+
+    h2o_mruby_config_vars_t config = ctx->handler->config;
+    mrb_value conf = mrb_hash_new_capa(mrb, 3);
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "code")), mrb_str_new(mrb, config.source.base, config.source.len));
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "file")), mrb_str_new(mrb, config.path, strlen(config.path)));
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "line")), mrb_fixnum_value(config.lineno));
+
+    /* run code and generate handler */
+    mrb_value result =
+        mrb_funcall(mrb, mrb_obj_value(mrb->kernel_module), "_h2o_prepare_app", 1, conf);
+    assert(mrb_array_p(result));
```

Comment:
I believe that we cannot have the assertion here.

My understanding that we need to check test if `mrb->exc` is NULL or not and take appropriate actions (which is done in the caller) before asserting that the value being returned is an array.